### PR TITLE
Vulkan: Fix CanvasItem::use_parent_material

### LINKED ISSUE #48861: Vulkan: Use Parent Material doesn't work
**Godot version:**

1450a72

**Steps to reproduce:**
1. Create a 2D scene with a Sprite2D child
2. Assign any material to the root, e.g. this invert shader
```
shader_type canvas_item;

void fragment() {
	COLOR.rgb = vec3(1) - texture(TEXTURE, UV).rgb;
}
```
3. Enable "Use Parent Material" in the sprite
4. Expect that the sprite will use the above-attached shader
5. You will get disappointed

### PR DESCRIPTION
**Pull request by TechnoPorg**
Fixes #48861
The "Use Parent Material" option now does something when enabled on a CanvasItem. As before, it's not just limited to a node's direct parent but can move up the tree until it finds a material.
I also corrected a typo in rendering_device_vulkan.h that didn't merit its own commit.

### CODE REVIEW

**Comment by clayjohn** [comment]
I'm not certain that this is the most efficient way to search the material owners. I'm comparing to how it is done in 3.x e.g. material owners are assigned in the VisualServer as the renderer traverses the tree. 

I imagine this approach (searching vertically through the tree) will start to become costly when there are long chains of "use parent material".

Before merging this, I think we need to explore if a solution like in 3.x is possible.

**Comment by TechnoPorg** [comment]
It seems that `RendererCanvasCull` is already assigning the root of the "use parent material" chain as the material owner for all the rest of the chain when it renders the canvas, so no vertical tree search is necessary. Is this what you mean or were you thinking it would be better to set the material owner only once when `use_parent_material` is changed @clayjohn?

**Review by clayjohn — APPROVED** [review]
Looks great! 

Sorry, I missed that you had forced pushed changes,

**Comment by TechnoPorg** [comment]
No problem, thanks for reviewing!

**Comment by clayjohn** [comment]
@JFonS @BastiaanOlij This is ready to merge if either one of you has a moment :)

**Review by JFonS — APPROVED** [review]

**Comment by JFonS** [comment]
Thanks!

**Pull request merged**
